### PR TITLE
Groundwork setup

### PR DESCRIPTION
**Pull request by SzyJ**
Add Premake setup with dependencies
Add code style configs
Configure project:
 - Add PCH
 - Write entry point in engine
 - Add logging system

### CODE REVIEW

File: Untitled-Engine/Src/Untitled/Core/Core.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2020 [Untitled Games]
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE', which is part of this source code package.
+// 
+// [+ :]
+
+#pragma once
+
+#ifndef UE_PLATFORM_WINDOWS
```
Comment (SzyJ) [inline Untitled-Engine/Src/Untitled/Core/Core.h]: ifdef redundant. WIll delete

File: README.md [hunk 2]
```diff
@@ -1,2 +1,22 @@
...
+git clone https://github.com/Untitled-Games/Untitled-Engine
+cd Untitled-Engine
+git submodule init
+git submodule update
+```
+
+## Premake
+This project uses [premake](https://github.com/premake/premake-core) to generate desired project files.
+
+```generate_vs2017.bat``` or ```generate_vs2019.bat``` scripts can be used to generate a Visual Studio solution.
```
Comment (Kney-Delach) [inline README.md]: Rename generation scripts to camel case and reference updated names here.

File: Untitled-Engine/Src/Untitled/Core/Log.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2020 [Untitled Games]
```
Comment (Kney-Delach) [inline Untitled-Engine/Src/Untitled/Core/Log.cpp:1]: Maybe move this file into its own directory in Core (Untitled-Engine/Src/Untitled/Core/Logger/Log.cpp) same goes for the header.

File: Untitled-Engine/Src/Untitled/Core/Log.h [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+}
+
+#ifdef UE_DEBUG
+
+#define UE_LOG_TRACE(...)    ::ue::Log::GetCoreLogger()->trace(__VA_ARGS__)
+#define UE_LOG_INFO(...)     ::ue::Log::GetCoreLogger()->info(__VA_ARGS__)
+#define UE_LOG_WARN(...)     ::ue::Log::GetCoreLogger()->warn(__VA_ARGS__)
+#define UE_LOG_ERROR(...)    ::ue::Log::GetCoreLogger()->error(__VA_ARGS__)
+#define UE_LOG_FATAL(...)    ::ue::Log::GetCoreLogger()->fatal(__VA_ARGS__)
```
Comment (Kney-Delach) [inline Untitled-Engine/Src/Untitled/Core/Log.h]: GetCoreLogger()->fatal doesn't exist, use GetCoreLogger()->critical instead, and rename macros to CRITICAL.

File: Untitled-Engine/Src/Untitled/Core/Log.h [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+#define UE_APP_TRACE(...)    ::ue::Log::GetAppLogger()->trace(__VA_ARGS__)
+#define UE_APP_INFO(...)     ::ue::Log::GetAppLogger()->info(__VA_ARGS__)
+#define UE_APP_WARN(...)     ::ue::Log::GetAppLogger()->warn(__VA_ARGS__)
+#define UE_APP_ERROR(...)    ::ue::Log::GetAppLogger()->error(__VA_ARGS__)
+#define UE_APP_FATAL(...)    ::ue::Log::GetAppLogger()->fatal(__VA_ARGS__)
```
Comment (Kney-Delach) [inline Untitled-Engine/Src/Untitled/Core/Log.h]: GetCoreLogger()->fatal doesn't exist, use GetCoreLogger()->critical instead, and rename macros to CRITICAL.

File: Untitled-Engine/Src/Untitled/Core/Log.h [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+#else
+
+#define UE_LOG_TRACE(...)
+#define UE_LOG_INFO(...)
+#define UE_LOG_WARN(...)
+#define UE_LOG_ERROR(...)
+#define UE_LOG_FATAL(...)
```
Comment (Kney-Delach) [inline Untitled-Engine/Src/Untitled/Core/Log.h]: Same as above.

File: Untitled-Engine/Src/Untitled/Core/Log.h [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+#define UE_APP_TRACE(...)
+#define UE_APP_INFO(...)
+#define UE_APP_WARN(...)
+#define UE_APP_ERROR(...)
+#define UE_APP_FATAL(...)
```
Comment (Kney-Delach) [inline Untitled-Engine/Src/Untitled/Core/Log.h]: Same as above.

File: generate_vs2017.bat [hunk 8]
```diff
@@ -0,0 +1,2 @@
+call Configuration\Premake\Binary\premake5.exe vs2017
```
Comment (Kney-Delach) [inline generate_vs2017.bat]: Rename this file to CamelCase.

File: generate_vs2019.bat [hunk 9]
```diff
@@ -0,0 +1,2 @@
+call Configuration\Premake\Binary\premake5.exe vs2019
```
Comment (Kney-Delach) [inline generate_vs2019.bat]: Rename this file to CamelCase.

File: premake5.lua [hunk 10]
```diff
@@ -0,0 +1,40 @@
+-- Copyright (c) 2020 [Untitled Games]
```
Comment (Kney-Delach) [inline premake5.lua]: Rename this file to CamelCase.

**Review by Kney-Delach — COMMENTED** [review]
View and respond to comments for merge approval.

**Review by Qozul — COMMENTED** [review]
Resolve other comments and I'm content.

File: Untitled-Engine/Src/Untitled/Core/Log.cpp (hunk 3 above)
Comment (SzyJ) [reply Untitled-Engine/Src/Untitled/Core/Log.cpp:1]: I think that the core directory will not have too many files anyway so didn't want to seperate it out more than is necessary. Do you think we will be expanding the logging functionality to include more than just these 2 files in the future?

Comment (Qozul) [reply Untitled-Engine/Src/Untitled/Core/Log.cpp:1]: It is possible we would want to separate out debug and error logging, but it doesn't seem necessary. In either case we could update this in the future if we do decide to expand it.

Comment (SzyJ) [reply Untitled-Engine/Src/Untitled/Core/Log.cpp:1]: I agree. If we need to extend in the future, we can separate it out more

File: README.md (hunk 2 above)
Comment (SzyJ) [reply README.md]: Done!

**Comment by SzyJ** [comment]
All comments resolved. Merging with master

**Pull request merged**